### PR TITLE
add ipoib to builder and own types

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: test/lan_test.rb [hunk 1]
```diff
@@ -357,8 +357,8 @@ def expect_modification_succeedes(modname, method)
       end
 
       it "does not configure the interface if it is not connected" do
-        allow(Yast::Lan).to receive(:connected_and_bridgeable?).with(anything).and_return(false)
-        expect(Yast::Lan).not_to receive(:ProposeItem)
+        allow(Yast::Lan).to receive(:connected_and_bridgeable?).with(anything, anything, anything).and_return(false)
```
Comment (mchf) [inline test/lan_test.rb]: ```any_args``` doesn't fit?

Comment (jreidinger) [reply test/lan_test.rb]: anything match single param, not list of params. In fact maybe better to remove whole "with".

**Comment by teclator** [comment]
For me it looks good

File: test/lan_test.rb (hunk 1 above)
Comment (mchf) [reply test/lan_test.rb]: well I thought that having any_args is shorter than having anything 3 times. But removing with completely makes the same service in this case.

File: src/lib/y2network/interface_config_builder.rb [hunk 2]
```diff
@@ -29,6 +29,18 @@ module Y2Network
   # Collects data from the UI until we have enough of it to create a {Y2Network::Interface}.
   # {Yast::LanItemsClass#Commit Yast::LanItems.Commit(builder)} uses it.
   class InterfaceConfigBuilder
+    include Yast::Logger
+
+    def self.for(type)
```
Comment (imobachgs) [inline src/lib/y2network/interface_config_builder.rb]: Just add some documentation here, please.

File: src/lib/y2network/interface_config_builder.rb [hunk 3]
```diff
@@ -29,6 +29,18 @@ module Y2Network
...
+      require "y2network/interface_config_builders/#{type}"
+      InterfaceConfigBuilders.const_get(type.to_s.capitalize).new
+    rescue LoadError => e
+      log.info "Specialed builder for #{type} not found. Fallbacking to default. #{e.inspect}"
+      res = new
```
Comment (imobachgs) [inline src/lib/y2network/interface_config_builder.rb]: Why not to pass the `type` in the constructor?

**Pull request merged**